pkg/polaris/polaris_test.go

```diff
@@ -748,3 +751,372 @@ func TestGcpPermissions(t *testing.T) {
+// and Bronze.
+//
+// To run this test against a Polaris instance the following environment
+// variables needs to be set:
+//   * TEST_INTEGRATION=1
+//   * RUBRIK_POLARIS_SERVICEACCOUNT_FILE=<path-to-polaris-service-account-file>
+func TestListK8sNamespace(t *testing.T) {
+	//if !boolEnvSet("TEST_INTEGRATION") {
+	//	t.Skipf("skipping due to env TEST_INTEGRATION not set")
+	//}
+
+	testClusterID := uuid.MustParse("a840e205-ac36-408d-9a12-7690769aaa88")
+	ctx := context.Background()
+	testServiceAccount := ServiceAccount{
+		ClientID:       "client|sIIw3uAxHqFsn3kUR78AUf1zMewyLB7p",
+		ClientSecret:   "WnmUX2luK5X_TcrMMzZUrFh-mU7gWWti0VS90onJ_uwygXsYajUwVOlWE1MArIs_",
+		Name:           "test",
+		AccessTokenURI: "https://manifest.dev-045.my.rubrik-lab.com/api/client_token",
+	}
+	// Load configuration and create client. Usually resolved using the
+	// environment variable RUBRIK_POLARIS_SERVICEACCOUNT_FILE.
+	//polAccount, err := DefaultServiceAccount(true)
+	//if err != nil {
+	//	t.Fatal(err)
+	//}
+
+	client, err := NewClient(ctx, &testServiceAccount, &polaris_log.DiscardLogger{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	nss, err := client.K8s().ListK8sNamespace(ctx, testClusterID)
+	if err != nil {
+		t.Fatal(err)
+	}
+	for _, ns := range nss {
+		fmt.Printf("%v\n", ns)
+	}
+
+}
+
+func TestUnassignSLA(t *testing.T) {
+	//if !boolEnvSet("TEST_INTEGRATION") {
+	//	t.Skipf("skipping due to env TEST_INTEGRATION not set")
+	//}
+
+	testNSID := uuid.MustParse("3b3d22b7-385c-5865-bd8f-0ff7534db42b")
+	ctx := context.Background()
+	testServiceAccount := ServiceAccount{
+		ClientID:       "client|YVaYtseZQXRSiwEHV8HaRKfiRfsU0BhX",
+		ClientSecret:   "BS0-2olXRhY51R1AW0qj9gdgYBOs6x4uJbUg-7DXIxQImigEiavo819R0ZTPwq8a",
+		Name:           "np-test",
+		AccessTokenURI: "https://demo.dev-017.my.rubrik-lab.com/api/client_token",
+	}
+	client, err := NewClient(ctx, &testServiceAccount, &polaris_log.DiscardLogger{})
+	if err != nil {
+		t.Fatal(err)
+	}
+	oks, err := client.Core().AssignSlaForSnappableHierarchies(
+		ctx,
+		nil,
+		core.NoAssignment,
+		[]uuid.UUID{testNSID},
+		[]core.SnappableLevelHierarchyType{},
+		true,  // shouldApplyToExistingSnapshots
+		false, // shouldApplyToNonPolicySnapshots
+		core.RetainSnapshots,
+	)
+	// Load configuration and create client. Usually resolved using the
+	// environment variable RUBRIK_POLARIS_SERVICEACCOUNT_FILE.
+	//polAccount, err := DefaultServiceAccount(true)
+	if err != nil {
+		t.Fatal(err)
+	}
+	t.Logf("%v\n", oks)
+
+}
+
+// TestTakeK8NamespaceSnapshot verifies that the SDK can take a
+// namespace snapshot
+func TestTakeK8NamespaceSnapshot(t *testing.T) {
+	testClusterID := uuid.MustParse("a840e205-ac36-408d-9a12-7690769aaa88")
+	ctx := context.Background()
+	testServiceAccount := ServiceAccount{
+		ClientID:       "client|2YDsG5FIYWQ7OW8AiRqA2tPwQEwhHpfU",
+		ClientSecret:   "ZLeoEHA6dvDGciAO1bFIIWJmdjlLLawFa6WC8IdpBfgphecsUQYB4pAOKovQbt4e",
```

Comment:
:exclamation:**Cycode**: Secret of type: 'Base64 Entropy' was found.
Severity: Medium <!-- 55d11bd4-cd17-4ba6-b4b7-0993b08c3e13 -->